commitment/taproot.go

```diff
@@ -150,11 +178,33 @@ func (t *TapscriptPreimage) TapHash() (*chainhash.Hash, error) {
 // VerifyNoCommitment verifies that the preimage is not a Taproot Asset
 // commitment.
 func (t *TapscriptPreimage) VerifyNoCommitment() error {
-	if IsTaprootAssetCommitmentScript(t.SiblingPreimage) {
-		return ErrPreimageIsTapCommitment
-	}
+	switch {
+	// A preimage smaller than a valid Taproot Asset commitment script needs
+	// no further inspection.
+	case len(t.SiblingPreimage) < TaprootAssetCommitmentScriptSize:
+		return nil
+
+	case len(t.SiblingPreimage) == TaprootAssetCommitmentScriptSize:
+		if IsTaprootAssetCommitmentScript(t.SiblingPreimage) {
+			return ErrPreimageIsTapCommitment
+		}
 
-	return nil
+		return nil
+
+	default:
+		// The sibling may be an encoded tapleaf; remove the version
```

Comment:
what is an "encoded tapleaf", i don't know that terminology

Reply:
The byte slice `leafVersion || compactSizeof(script) || script`, which is the input for `h_tapleaf`.

Reply:
This function needs to be made type safe, TapScriptPreimage should only ever be an encoded tapleaf or an unencoded one, and that should be handled before this is reached.